src/async-wrap.cc

```diff
@@ -282,37 +285,87 @@ bool AsyncWrap::EmitAfter(Environment* env, double async_id) {
 class PromiseWrap : public AsyncWrap {
  public:
   PromiseWrap(Environment* env, Local<Object> object, bool silent)
-    : AsyncWrap(env, object, PROVIDER_PROMISE, silent) {}
+    : AsyncWrap(env, object, PROVIDER_PROMISE, silent) {
+    MakeWeak(this);
```

Comment:
@addaleax 
> The thing is, MakeWeak is a not the best choice of name. What it actually does is to set the 0th internal field of the associated object to the passed pointer, and then mark the handled contained in the Wrap object as weak and set a destroy callback for V8 to call.

And... I'm to blame for this one. In fe2df3b842205d8dedffcc83de387d43579ec760 I carefully went through to make sure `Wrap()` and `ClearWrap()` were called for all applicable instances forgetting that 3.5 years ago for whatever forsaken reason when I implemented `BaseObject` (d120d92b) I made `MakeWeak()` automatically `Wrap()` the object. This was a horrible mistake on my part

Basically I would be happy if we removed the call to `Wrap()` in `BaseObject::MakeWeak()`.

@JiaLiPassion 
 I just read MakeWeak API and node_object_wrap.h, and I will continue to learn about it.

Be careful. AsyncWrap doesn't use `node_object_wrap`. It uses `base-object`